include/boost/beast/http/impl/fields.hpp

```diff
@@ -590,8 +584,8 @@ basic_fields<Allocator>::
 set(field name, string_view const& value)
 {
     BOOST_ASSERT(name != field::unknown);
-    set_element(new_element(name, to_string(name),
-        static_cast<string_view>(value)));
+    set_element(
```

Comment:
`set_element` should receive a pointer to keep things consistent

Reply:
`set_element`, `insert_element`, and `delete_element` receive references because they don't handle `nullptr`. I believe changing `new_element` to return pointers isn't a good idea either because it never returns `nullptr`. I think changing the name of the other `new_element` overload would be a better option, something like `try_create_new_element` or `try_new_element`.

Reply:
`*_element` are private functions so we can do whatever we want. The rules are relaxed. We should do what is easiest for maintenance. My opinion, pointers are better than having new names and different return types. But it is up to you.